### PR TITLE
be 2.13 friendly

### PR DESCRIPTION
**Pull request by SethTisue**
* treat 2.13 and 2.12 the same, for the time being, in scalaVersion
  handling in the build
* `Pair` is a deprecated type alias for `Tuple2` and was removed in 2.13
* temporary workaround (marked TODO) for removal of deprecated
  `conforms` from scala.Predef; Akka team to substitute a proper fix;
  details at https://github.com/akka/akka/issues/22451

### CODE REVIEW

**Comment by SethTisue** [comment]
this makes Akka work again in the Scala 2.13 community build.

note this should not be merged as-is, but should be combined with a proper fix for #22451, instead of the kludge I put in

**Comment by ktoso** [comment]
Thanks! I fixed the klunge in your branch right away :-)
https://github.com/akka/akka/pull/22486/commits/daefa4b3ea42f5d32da3fae6f7d9163b469ca659

> Pair is a deprecated type alias for Tuple2 and was removed in 2.13

Just sanity checking: this is binary compatible for 2.12? I mean, this change will affect our 2.11 and 2.12 artifacts for the time being, should this just work?

**Comment by SethTisue** [comment]
> this is binary compatible for 2.12

correct. type aliases are a compile-time-only fiction

**Comment by ktoso** [comment]
And the alias was there in 2.11 too? (since we cross release for 2.11 and 2.12)

**Comment by SethTisue** [comment]
`Tuple2` and `Pair` are both ancient. the real class is `Tuple2`, `Pair` was the alias

**Comment by ktoso** [comment]
Oh, thanks for confirming :) Looks all good then

**Comment by SethTisue** [comment]
I'll rebase to make the history a little cleaner

**Review by ktoso — APPROVED** [review]

File: akka-stream/src/main/scala/akka/stream/javadsl/SubFlow.scala [hunk 1]
```diff
@@ -1170,7 +1170,7 @@ class SubFlow[-In, +Out, +Mat](delegate: scaladsl.SubFlow[Out, Mat, scaladsl.Flo
    *
    * '''Cancels when''' downstream cancels
    */
-  def zip[T](source: Graph[SourceShape[T], _]): SubFlow[In, Out @uncheckedVariance Pair T, Mat] =
+  def zip[T](source: Graph[SourceShape[T], _]): SubFlow[In, Out @uncheckedVariance Tuple2 T, Mat] =
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/javadsl/SubFlow.scala]: @ktoso hm, shouldn't this be `akka.japi.Pair`
(unrelated to this PR)

Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/javadsl/SubFlow.scala]: Whoa, bug in our API :< This should be the japi all the time but wasn't :<

File: akka-stream/src/main/scala/akka/stream/javadsl/SubSource.scala [hunk 2]
```diff
@@ -1163,7 +1163,7 @@ class SubSource[+Out, +Mat](delegate: scaladsl.SubFlow[Out, Mat, scaladsl.Source
    *
    * '''Cancels when''' downstream cancels
    */
-  def zip[T](source: Graph[SourceShape[T], _]): SubSource[Out @uncheckedVariance Pair T, Mat] =
+  def zip[T](source: Graph[SourceShape[T], _]): SubSource[Out @uncheckedVariance Tuple2 T, Mat] =
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/javadsl/SubSource.scala]: same here https://github.com/akka/akka/pull/22486/files#r104931037

**Comment by ktoso** [comment]
Will merge and follow up by fixing the Pair

File: project/Dependencies.scala [hunk 3]
```diff
@@ -21,13 +21,17 @@ object Dependencies {
     scalaVersion := System.getProperty("akka.build.scalaVersion", crossScalaVersions.value.head),
     scalaStmVersion := sys.props.get("akka.build.scalaStmVersion").getOrElse("0.8"),
     scalaCheckVersion := sys.props.get("akka.build.scalaCheckVersion").getOrElse(
-      if (scalaVersion.value.startsWith("2.12")) "1.13.4" // does not work for 2.11
-      else "1.13.2"
+      CrossVersion.partialVersion(scalaVersion.value) match {
```
Comment (dwijnand) [inline project/Dependencies.scala]: 👍

**Pull request merged**